Don't require an optical drive for stub file handling

Woerd88 opened the pull request:
Partially Reverted: xbmc@731a7b1
Forum discussion: http://forum.kodi.tv/showthread.php?tid=178014&page=2&highlight=stub

With this change ppl can check their stub information on systems that don't have an optical drive.
(Raspberry PI's, Android Tablets etc.)

Edit: Sorry about the lame default title, i did not create it properly.
I am still learning to work with git.

<!--- Provide a general summary of your change in the Title above -->

## Description
<!--- Describe your change in detail -->
Removed HasOpticalDrive check so we can see the custom information from the stub file.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->
If fixes the issue where you can't "play" a stub file on a device where no optical drive is present.
Therefor you can't view custom stubfile messages, e.g. "Enter disc 46".

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->
Not at all! I am a developer but have no expierence with building kodi (so far)
As @da-anda proposed, maybe someone can launch an automatic build?
I am having difficulty reading/following the checklist because of the server/wiki issues. 
(Cloudfare switch)

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Razzeee commented:
First, thank you for your contribution and sorry for it sitting here so long.

There has been an alternative implementation, from @dml33 it would be nice if you two could speak with each other and come to terms, on who is doing this patch. Thank you! :)

Rechi commented:
@Woerd88 this needs a rebase

Woerd88 committed:
Don't require an optical drive for stub file handling

Paritally Reverted: xbmc@731a7b1
Forum discussion: http://forum.kodi.tv/showthread.php?tid=178014&page=2&highlight=stub

With this change ppl can check their stub information on systems that don't have an optical drive.
(Raspberry PI's, Android Tablets etc.)

phunkyfish commented:
PR closed as it appears to be obsolete. 

If the author would like it kept open please just request so on the PR. If you feel there is still merit in merging this PR you can request the label be changed to `PR Cleanup: Backburner`.
